src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,636 @@
+ *     Block1 result: {2,3,6,7}
+ *     Block2 result: {2,0,6,4}
+ *
+ *     The next step is to convert these ids into their binary representation:
+ *     Block1 result: {010,011,110,111}
+ *     Block2 result: {010,000,110,100}
+ *
+ *     Note that we can now read off the shared face relative to each Block
+ * easily:
+ *     Block1 result: Upper eta (All binary Block1 ids have a 1 in the second
+ * position)
+ *     Block2 result: Lower xi (All binary Block2 ids have a 0 in the 3rd
+ * position)
+ *
+ *     Now we know that Direction<3>::upper_eta() in Block1 corresponds to
+ * Direction<3>::lower_xi().opposite()
+ * in Block2.
+ *     ".opposite()" is a result of the Directions to a Block face being
+ * anti-parallel
+ *     because each Block lies on the opposite side of the shared face.
+ *
+ *    The remaining two correspondences are given by the Alignment of the shared
+ * face.
+ *    It is useful to know the following information:
+ *    In the Local CNS, if an edge lies along the xi direction, if one takes the
+ * two corners making up that edge and takes the
+ *    difference of their ids, one always gets the result +/- 1.
+ *    Similarly, if the edge lies in the eta direction, the result will be +/-
+ * 2.
+ *    Finally, if the edge lies in the zeta direction, the result will be +/- 4.
+ *    We use this information to determine the Alignment of the shared face:
+ *
+ *    Block1: 3-2=1  => This edge is in the +xi direction.
+ *    Block2: 0-2=-2 => This edge is in the -eta direction.
+ *    Then, +xi in Block1 corresponds to -eta in Block2.
+ *
+ *    Block1: 6-2=4 => This edge is in the +zeta direction.
+ *    Block2: 6-2=4 => This edge is in the +zeta direction.
+ *    Then, +zeta in Block1 corresponds to +zeta in Block2.
+ *
+ *    The corresponding directions in each Block have now been deduced.
+ *
+ *    To confirm, we can use the other ids as well and arrive at the same
+ * result:
+ *    Block1: 7-6-1  => +xi
+ *    Block2: 4-6=-2 => -eta
+ *
+ *    Block1: 7-3=4  => +zeta
+ *    Block2: 4-0=4  => +zeta
+ *
+ *  ----------------------------------------------------------------------
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <numeric>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Orientation.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+template <size_t VolumeDim>
+struct DomainHelpers {
+
+  /// \returns the common corners in the global corner numbering scheme.
+  static std::vector<size_t> get_common_global_corners(
+      const std::array<size_t, two_to_the(VolumeDim)>& block_1_global_corners,
+      const std::array<size_t, two_to_the(VolumeDim)>& block_2_global_corners);
+
+  /// \returns the common corners in the local corner numbering scheme.
+  template <typename Seq>
```

Comment:
It is a vector when we dont know the size of block_global_common_corners (in the case of finding common corners in our domain, it can be between 0 and 4 for 3D)
It is an array when we are setting up periodic boundary conditions (when we have 4 for 3D)

Reply:
Okay, I wonder if it would make more sense to just use a `vector` always and then not have this be a template. An open-ended question for others to weigh in on :)